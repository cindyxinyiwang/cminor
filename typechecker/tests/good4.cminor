// declarations

hello1: function integer () = {
	print "hello world";
}

x: integer = 1;
y: boolean = true;
y1: boolean = false;

// array [] only allowed in param
random: function array [1] integer (x: integer, y: integer) = {
	z: integer = x+y;
	a: array [3] integer = {1, 2,3};
	return a;
}

func: function array [10] integer (m: boolean, n: string, y: integer) = {
	print m, n, y;

	y = x;
	m = y1 || x==y;

	return random(1, 2);
}
a: array [10] integer = {1,2,3};

proto: function void (i: integer, j: boolean, k: string, p: char, m: array [] integer);

useProto: function void (p: integer, q:integer) = {
	proto(p, true, "hello", 'h', a);
	func(true, "hello", 1);
}
