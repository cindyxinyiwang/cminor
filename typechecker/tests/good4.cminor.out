x resovles to param 0
y resovles to param 1
a resovles to local 1
m resovles to param 0
n resovles to param 1
y resovles to param 2
y resovles to param 2
x resovles to global x
m resovles to param 0
y1 resovles to global y1
x resovles to global x
y resovles to param 2
random resovles to global random
p resovles to param 0
a resovles to global a
proto resovles to global proto
func resovles to global func
